zerver/lib/message.py

```diff
@@ -173,6 +173,9 @@ class SendMessageRequest:
     disable_external_notifications: bool = False
 
 
+# Used to check for valid string values for message `type` parameters.
+VALID_MESSAGE_TYPES = ["direct", "private", "stream"]
```

Comment:
I wasn't sure if there was another place in the backend code where it would be better to define this constant so that it can be shared by both `zerver/views/message-send.py` and `zerver/views/typing.py`?

Reply:
I suppose `Message.API_RECIPIENT_TYPES` is the other place I'd consider, but that might be confusing. I think we should call these "recipient types", not "message types", because we have considered making "message type" be a phase reserved for special types of messages, like polls and certain types of special Notification Bot messages.

Reply:
Okay. I moved this to `Message.API_RECIPIENT_TYPES` and added a comment. I used the comments for other `API_` constants in `models.py` as a jumping off point for that comment, but it may need some revision.